### PR TITLE
Add Travis, Codecov

### PR DESCRIPTION
**Pull request by arp242**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,13 @@
 [![Build Status](https://travis-ci.org/Teamwork/httperr.svg?branch=master)](https://travis-ci.org/Teamwork/httperr) [![Codecov](https://img.shields.io/codecov/c/github/Teamwork/httperr.svg?style=flat)](https://codecov.io/gh/Teamwork/httperr) [![GoDoc](https://godoc.org/github.com/Teamwork/httperr?status.svg)](http://godoc.org/github.com/Teamwork/httperr)
+[![codecov](https://codecov.io/gh/Teamwork/httperr/branch/master/graph/badge.svg)](https://codecov.io/gh/Teamwork/httperr)
+[![GoDoc](https://godoc.org/github.com/Teamwork/httperr?status.svg)](https://godoc.org/github.com/Teamwork/httperr)
```
Comment (flimzy) [inline README.md]: These badges already exist on the line above.

File: bin/coverage [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
```
Comment (flimzy) [inline bin/coverage]: I wonder if it's a good idea to use this file for these small packages... all the complication is due to the way we run retro-fitted tests in Desk, and can be completely replaced with a single line in most cases.  Even on larger internal projects, I expect we'll likely never need Desk's complication again, as long as we start the projects with tests.

File: bin/start [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+# Verbose output?
+verbose=${verbose:-0}
+
+
+###
+# For the following the defaults should almost always be okay. If not, then
+# consider fixing the project maybe?
+
+# Binary name.
```
Comment (flimzy) [inline bin/start:29]: This comment would probably be more accurate as `Package name`, although that's ambiguous with the following comment...

**Comment by flimzy** [comment]
I like the idea of this CI stuff being easily bundled, but it kind of feels like unnecessary complication for most of these small packages.  The only part that seems very useful is the linter, and even that is probably way over-complicated, since we can get rid of all the "incremental improvement" features on these new projects.

What do you think?

**Comment by arp242** [comment]
I considered simplifying them, but:

1.  I like having the standard `./bin/lint`, `./bin/test`, etc. locations. 
2. Sure, they could be simpler, but as far as I see, there's no real harm in having slightly more complex tools, either? Especially since:
3. We might need the complexity in the future. For e.g. the `./bin/coverage` script you'll need it when a package grows subpackages. I don't necessarily expect that any time soon for `httperr`, but it's not inconceivable either.

Personally, I care the most about the first point. If you want to simplify things then be my guest.

File: bin/start (hunk 3 above)
Comment (arp242) [reply bin/start:29]: Actually the way it's used "Docker image name" is more accurate:

    docker build -t "teamwork/$name:$docker_tag" .
    docker push "teamwork/$name:$docker_tag"

Comment (flimzy) [reply bin/start:29]: That's not very accurate for these libs, either, since they don't use docker.

**Comment by flimzy** [comment]
> I like having the standard ./bin/lint, ./bin/test, etc. locations.

I'm not disagreeing with standard locations, or standard scripts. I just think the scripts are overkill for most of these projects.  When they're useful (in large, monolithic projects such as Desk, possibly Projects/CRM/HelpDocs/etc, as those grow/move to go), I think your design is great.

I just question whether there's enough overlap between those cases and a simple lib like this, to actually use the same tool.

> there's no real harm in having slightly more complex tools

But it's not "slightly more complex."  The difference between `go test -race -cover -covermode=set ./... -coverprofile=coverage.txt && bash <(curl -s https://codecov.io/bash)` and the "standard" test/coverage scripts is 158 lines of complexity. 158x more complex isn't "slightly" more complex. It's immensely more complex.

> We might need the complexity in the future

Then lets add it in the future, if/when we need it.  I seriously doubt we will, but if we do, it's just as easy to add it then as it is now.

**Comment by arp242** [comment]
Yeah sure, that's a valid reasoning. I simplified the `test` and `coverage` scripts. For the `lint` script you still sort of need most of it, though.

File: README.md (hunk 1 above)
Comment (arp242) [reply README.md]: Oh, because it was all on a single line I thought it was just the Travis one >_<

**Comment by flimzy** [comment]
> For the lint script you still sort of need most of it, though.

Not really. 99% of the linter complexity is for dealing with old code bases, where we don't want to clean it all up at once. For a pristine code base, all we need is:

    go get -u gopkg.in/alecthomas/gometalinter.v1 ...
    gometalinter.v1 --deadline=60s --vendor --disable-all --enable=...

Which is what https://github.com/Teamwork/sqluster/pull/3 provides for the sqluster repo.

**Pull request closed without merging**